ml-agents/mlagents/trainers/components/reward_signals/reward_signal.py

```diff
@@ -0,0 +1,57 @@
+import logging
+from mlagents.trainers.trainer import UnityTrainerException
+from mlagents.trainers.policy import Policy
+
+import tensorflow as tf
+
+logger = logging.getLogger("mlagents.trainers")
+
+
+class RewardSignal(object):
```

Comment:
Worth using abstract base class (https://docs.python.org/3/library/abc.html) and making these `@abc.abstractmethod`'s?

Reply:
I did have it like that in an earlier iteration, but wasn't sure about it since it isn't a construct we use anywhere else in the code. Maybe it's worth bringing back?

Reply:
Your call - the other option is to provide a default implementation some of the methods, e.g. `return {}` in `update()`